cocos/base/ObjectFactory.cpp

```diff
@@ -105,17 +62,13 @@ void ObjectFactory::destroyInstance()
 Ref* ObjectFactory::createObject(const std::string &name)
 {
     Ref *o = nullptr;
-    do 
-    {
-        const TInfo t = _typeMap[name];
-        if (t._fun != nullptr)
-        {
-            o = t._fun();
-        }else if (t._func != nullptr)
-        {
-            o = t._func();
-        }
-    } while (0);
+    const TInfo t = _typeMap[name];
+    if (t._fun != nullptr) {
+        o = t._fun();
```

Comment:
Consider change to return and remove "o" variable

Reply:
Done.

I would like to change `_fun` member in TInfo in a separate PR. All changes made so far are limited to these two files. Changing `_fun` will involves more.